src/Matrix.jl

```diff
@@ -2873,6 +2873,265 @@ function rank(M::MatrixElem{T}) where {T <: FieldElement}
    return lu!(P, A)
 end
 
+@doc raw"""
+   rank_interpolation(M::MatrixElem{T}) where {T <: PolyRingElem} -> Int
+   rank_interpolation(M::MatrixElem{T}) where {T <: MPolyRingElem} -> Int
+   rank_interpolation(M::MatrixElem{T}) where {T <: AbstractAlgebra.Generic.RationalFunctionFieldElem} -> Int
+
+Returns the rank of $A$ using an interpolation-like method. 
+
+# Examples
+
+```jldoctest
+julia> Qx, x = polynomial_ring(QQ, :x);
+
+julia> rank_interpolation(matrix(Qx, 2, 2, [x 2x; x^2 1])
+2
+
+julia> Qy, y = rational_function_field(QQ, :y);
+
+julia> rank_interpolation(matrix(Qy, 2, 2, [1//y -y^2; 3 2-y])
+2
+```
+"""
+function rank_interpolation(M::MatrixElem{<: PolyRingElem{<: FieldElem}})
+   n = nrows(M)
+   m = ncols(M)
+   if is_zero(n) | is_zero(m)
```

Comment:
```suggestion
   if is_zero(n) || is_zero(m)
```
`||` is "lazy" in the sense that if the first statement is true, the second will not be executed. It is not a big difference here, but we use `||` everywhere.